tests/Install-Scripts.Test/GivenThatIWantToInstallTheSdkFromAScript.cs

```diff
@@ -0,0 +1,303 @@
+                ("master", "6\\.0\\.1.*"),
+            };
+
+        /// <summary>
+        /// All the distinct runtime options that will be tested together with their identifiers.
+        /// </summary>
+        private static readonly IReadOnlyList<(string runtime, string runtimeIdentifier)> _runtimes = new List<(string runtime, string runtimeIdentifier)>()
+        {
+            ("dotnet", "Microsoft.NETCore.App"),
+            ("aspnetcore", "Microsoft.AspNetCore.App"),
+            ("windowsdesktop", "Microsoft.WindowsDesktop.App"),
+        };
+
+        public static IEnumerable<object?[]> ChannelRuntimeTestCases
+        {
+            get
+            {
+                // Download SDK using branches as channels
+                foreach(var sdkBranchInfo in _sdkBranches)
+                {
+                    yield return new object?[]
+                    {
+                        sdkBranchInfo.branch,
+                        sdkBranchInfo.versionRegex,
+                        null,
+                        null,
+                    };
+                }
+
+                // Download runtimes using branches as channels
+                foreach(var runtimeBranchInfo in _runtimeBranches)
+                {
+                    foreach(var runtime in _runtimes)
+                    {
+                        yield return new object?[]
+                        {
+                            runtimeBranchInfo.branch,
+                            runtimeBranchInfo.versionRegex,
+                            runtime.runtime,
+                            runtime.runtimeIdentifier,
+                        };
+                    }
+                }
+
+                // Download SDK and runtimes from darc channels.
+                foreach(var channelInfo in _channels)
+                {
+                    yield return new object?[]
+                    {
+                        channelInfo.channel,
+                        channelInfo.versionRegex,
+                        null,
+                        null,
+                    };
+
+                    foreach (var runtimeInfoe in _runtimes)
+                    {
+                        yield return new object?[]
+                        {
+                            channelInfo.channel,
+                            channelInfo.versionRegex,
+                            runtimeInfoe.runtime,
+                            runtimeInfoe.runtimeIdentifier,
+                        };
+                    }    
+                }
+            }
+        }
+
+        /// <summary>
+        /// The directory that the install script will install the .NET into.
+        /// </summary>
+        private readonly string _sdkInstallationDirectory;
+
+        /// <summary>
+        /// Instantiates a GivenThatIWantToInstallTheSdkFromAScript instance.
+        /// </summary>
+        /// <remarks>This constructor is called once for each of the tests to run.</remarks>
```

Comment:
Is it called once for test class? or once for each test method? or once for each test case?

Reply:
Once for each test case. Is that what you though or should we update the comments?